src/main/java/org/mule/extension/sftp/internal/connection/SftpClient.java

```diff
@@ -376,12 +389,29 @@ public void write(String path, InputStream stream, FileWriteMode mode) throws IO
    */
   public String getHome() {
     try {
-      return session.executeRemoteCommand("pwd").trim();
+      return executePWDCommandWithTimeout();
```

Comment:
Save the value of the resolved `home` for later usage. Sync the access to the value.

```
synchronized(this.home) {
    if (this.home == null) {
        this.home = executePWDCommandWithTimeout();
    }
}
return this.home;
```